File: src/main/kotlin/com/dev/reCode/network/rest/controllers/RestApiController.java
package com.dev.reCode.network.rest.controllers;


import com.dev.reCode.network.dto.ConverterResponse;
import com.dev.reCode.network.rest.services.RestApiService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;


@RestController
@RequestMapping("api")
@AllArgsConstructor
public class RestApiController {
    private RestApiService restApiService;
    @PostMapping("/convert")
    public ConverterResponse handleSubmit(@RequestParam String inputData) {
        return restApiService.convert(inputData);
    }
}

File: src/main/kotlin/com/dev/reCode/network/rest/controllers/RestApiController.kt
package com.dev.reCode.network.rest.controllers

import com.dev.reCode.Vba2JsConverter
import com.dev.reCode.network.apiModels.ConverterResponse
import com.dev.reCode.network.rest.services.RestApiService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController


@RestController
@RequestMapping("api")
class RestApiController {
    val restApiService = RestApiService()
    @PostMapping("/convert")
    fun handleSubmit(@RequestParam inputData: String): ConverterResponse {
        return restApiService.convert(inputData)
    }
}